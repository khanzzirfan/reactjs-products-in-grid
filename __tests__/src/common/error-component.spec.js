import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ErrorComponent from 'src/common/error-component';

describe("error component test", () => {

    let wrapper = null;
    beforeEach(() => {
        wrapper = shallow(<ErrorComponent />);
    })

    it("component should be defined", () => {
        expect(wrapper).toBeDefined();
    });

    it("component should match to snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("it should render with correct class names", ()=>{
        expect(wrapper.find('.error-template').exists()).toBeTruthy();
        expect(wrapper.find('.error-details').exists()).toBeTruthy();

        expect(wrapper.find('.error-actions').exists()).toBeTruthy();

    });

    it("it should render with correct text", ()=>{
        expect(wrapper.text()).toContain("404 Not Found");
        expect(wrapper.text()).toContain("Sorry, an error has occured, Requested data not found!");
        expect(wrapper.text()).toContain("Oops!");
    });

});