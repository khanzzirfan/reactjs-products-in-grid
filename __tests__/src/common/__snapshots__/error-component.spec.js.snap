// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error component test component should match to snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12"
  >
    <div
      className="error-template"
    >
      <h1>
        Oops!
      </h1>
      <h2>
        404 Not Found
      </h2>
      <div
        className="error-details"
      >
        Sorry, an error has occured, Requested data not found!
      </div>
      <div
        className="error-actions"
      >
        <a
          className="btn btn-primary btn-lg"
          href="javascript:void(0)"
        >
          <span
            className="glyphicon glyphicon-home"
          />
          Take Me Home 
        </a>
        <a
          className="btn btn-default btn-lg"
          href="javascript:void(0)"
        >
          <span
            className="glyphicon glyphicon-envelope"
          />
           Contact Support 
        </a>
      </div>
    </div>
  </div>
</div>
`;
